integration/backward_compatibility_test.go

```diff
@@ -23,22 +24,26 @@ import (
 	"github.com/grafana/mimir/integration/e2emimir"
 )
 
-// previousVersionImages returns a list of previous image version to test backwards
-// compatibility against. If MIMIR_PREVIOUS_IMAGES is set to a comma separted list of image versions,
-// then those will be used instead of the default versions. Note that the overriding of flags
-// is not currently possible when overriding the previous image versions via the environment variable.
-func previousVersionImages() map[string]e2emimir.FlagMapper {
-	if overrideImageVersions := os.Getenv("MIMIR_PREVIOUS_IMAGES"); overrideImageVersions != "" {
+// previousVersionImages returns a list of previous image version to test backwards compatibility against.
+// If MIMIR_PREVIOUS_IMAGES is set to a comma separted list of image versions,
+// then those will be used instead of the default versions.
+// If MIMIR_PREVIOUS_IMAGES is set to a JSON, it supports mapping flags for those versions,
+// see TestParsePreviousImageVersionOverrides for the JSON format to use.
+func previousVersionImages(t *testing.T) map[string]e2emimir.FlagMapper {
+	if overrides := previousImageVersionOverrides(t); len(overrides) > 0 {
 		previousVersionImages := map[string]e2emimir.FlagMapper{}
 
-		// Overriding of flags is not currently supported when overriding the list of images.
-		for _, image := range strings.Split(overrideImageVersions, ",") {
-			previousVersionImages[image] = e2emimir.ChainFlagMappers(
+		for _, override := range overrides {
+			mappers := []e2emimir.FlagMapper{
 				cortexFlagMapper,
 				revertRenameFrontendToQueryFrontendFlagMapper,
 				ingesterRingRename,
 				ingesterRingNewFeatures,
```

Comment:
I couldn't find `MIMIR_PREVIOUS_IMAGES` set in this repository, so I wonder if these still make sense here, or we should move this responsibility to whoever is defining `MIMIR_PREVIOUS_IMAGES` and stop assuming the flags they need here.

WDYT @pracucci ?

Reply:
I think it was used in GEM. Before this PR there was no way to specify per-image remappings, so that's why we were doing it here. You can do whatever change you want here, as far as it works for GEM.

Reply:
Then I'll just remove them from here, and I'll define it there, since only GEM knows what flags it need to remap.